src/core/tsi/ssl_transport_security.cc

```diff
@@ -1410,6 +1407,36 @@ static void ssl_handshaker_destroy(tsi_handshaker* self) {
   gpr_free(impl);
 }
 
+// Removes the bytes remaining in |impl->SSL|'s read BIO and writes them to
+// |bytes_remaining|.
+static tsi_result ssl_bytes_remaining(tsi_ssl_handshaker* impl,
+                                      unsigned char** bytes_remaining,
+                                      size_t* bytes_remaining_size) {
+  if (impl == nullptr || bytes_remaining == nullptr ||
+      bytes_remaining_size == nullptr) {
+    return TSI_INVALID_ARGUMENT;
+  }
+  // Atempt to read all of the bytes in SSL's read BIO. These bytes should
+  // contain (encrypted) application data that was appended to a ClientFinished
+  // or ServerFinished record.
+  size_t bytes_in_ssl = BIO_pending(SSL_get_rbio(impl->ssl));
```

Comment:
Isn't it `BIO_pending(impl->network_io)`?

Reply:
That's what I though originally too, but @agl explained to me that it should be the read BIO held by `impl->ssl`. The reason is that there are [2 BIO's created as a pair](https://github.com/grpc/grpc/blob/eff1579811248a8117d721f3d6c5d2c7c5f3416b/src/core/tsi/ssl_transport_security.cc#L1511), so gRPC should write to `impl->network_io` and these are read from the read BIO held by `impl->ssl`.

Reply:
Thanks for clarification!